.github/workflows/mac.yml

```diff
@@ -0,0 +1,132 @@
+name: GraphScope CI on MacOS
+
+on:
+  # Trigger the workflow on push or pull request, but only for the main branch
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - 'gnn_engine/**'
+      - '**.md'
+      - '**.rst'
+      - 'docs/**'
+      - 'research/**'
+      - 'scripts/**'
+      - 'tutorials/**'
+  pull_request:
+    branches:
+      - main
+    paths-ignore:
+      - 'gnn_engine/**'
+      - '**.md'
+      - '**.rst'
+      - 'docs/**'
+      - 'research/**'
+      - 'scripts/**'
+      - 'tutorials/**'
+
+jobs:
+  build:
+    runs-on: macOS-latest
+    env:
+      JAVA_HOME: /Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home
+      OPENSSL_ROOT_DIR: /usr/local/opt/openssl
+      OPENSSL_LIBRARIES: /usr/local/opt/openssl/lib
+      OPENSSL_SSL_LIBRARY: /usr/local/opt/openssl/lib/libssl.dylib
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: true
+
+    - name: Setup tmate session
+      if: false
+      uses: mxschmitt/action-tmate@v2
+
+    - name: Install Dependencies for MacOS
+      run: |
+        pushd ${GITHUB_WORKSPACE}
+        brew install llvm@9
+
+        # setup envs
+        export CC=/usr/local/opt/llvm@9/bin/clang
+        export CXX=/usr/local/opt/llvm@9/bin/clang++
+        export LDFLAGS=-L/usr/local/opt/llvm@9/lib
+        export CPPFLAGS=-I/usr/local/opt/llvm@9/include
+        export PATH=/usr/local/opt/llvm@9/bin:$PATH
+        /bin/bash ./scripts/install_dependencies.sh
+        popd
+
+    - name: Install libgrape-lite
+      run: |
+        git clone -b master --single-branch --depth=1 https://github.com/alibaba/libgrape-lite.git
+        cd libgrape-lite
+        mkdir build && cd build
+        cmake ..
+        make -j
```

Comment:
Use `make -j$(nproc)` instead, otherwise the Github provided runner may hang up sometimes.

Reply:
it get ```command not found: nproc``` error in Mac

Reply:
`sysctl -n hw.physicalcpu` can do.